superset/assets/javascripts/explorev2/components/controls/Filter.jsx

```diff
@@ -100,7 +97,7 @@ export default class Filter extends React.Component {
           name="filter-value"
           value={filter.val}
           isLoading={this.state.valuesLoading}
-          choices={filter.choices}
+          choices={this.state.valueChoices}
```

Comment:
there's another `colChoices` variable somewhere in the in the code of this component that represent the column name choices. I decided to be a bit more specific since I was a tiny bit  confused while reading the code.